addons/font/bmfont.c

```diff
@@ -0,0 +1,498 @@
+#include "allegro5/allegro.h"
+#include "allegro5/allegro_font.h"
+#include <stdio.h>
+
+#include "allegro5/internal/aintern_font.h"
+
+#include "font.h"
+#include "xml.h"
+
+ALLEGRO_DEBUG_CHANNEL("font")
+
+typedef struct {
+   int first;
+   int second;
+   int amount;
+} BMFONT_KERNING;
+
+typedef struct {
+   int page;
+   int x, y;
+   int width, height;
+   int xoffset, yoffset;
+   int xadvance;
+   int chnl;
+   int kerning_pairs;
+   BMFONT_KERNING *kerning;
+} BMFONT_CHAR;
+
+typedef struct BMFONT_RANGE BMFONT_RANGE;
+
+struct BMFONT_RANGE {
+   int first;
+   int count;
+   BMFONT_CHAR **characters;
+   BMFONT_RANGE *next;
+};
+
+typedef struct {
+   ALLEGRO_USTR *tag;
```

Comment:
Maybe it'd be clearer if this mentioned that these were ephemeral for parsing... or maybe even there should be a dedicated structure for parsing.

Reply:
I moved it to a dedicated structure.